src/UserRemovalHandler.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * Copyright 2023 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\user\UserInterface;
+
+/**
+ * Default user removal handler service implementation.
+ *
+ * User removal can happen programmatically and when that happens Drupal
+ * developers may not always want to remove developer accounts from Apigee
+ * as that destroy their API keys too. This service enables developers to
+ * use the service decorator pattern to tweak if and how the developer
+ * removal happens when a user is deleted.
+ */
+class UserRemovalHandler implements UserRemovalHandlerInterface {
```

Comment:
Suggedted name `RemoveDeveloperWithUserUserRemovalHandler`.

Make  the class `final` mark the `entityTypeManager` property as `private`, and use native typehints.